src/ngraph/pass/nop_elimination.cpp

```diff
@@ -134,62 +134,264 @@ static bool eliminate_reshape_v1(const std::shared_ptr<Node>& node)
     return false;
 }
 
+static size_t count_unknown_dims(const PartialShape& ps)
+{
+    size_t rc = 0;
+    if (ps.is_static())
+    {
+        return rc;
+    }
+    for (auto i = 0; i < ps.rank().get_length(); i++)
+    {
+        if (ps[i].is_dynamic())
+        {
+            rc += 1;
+        }
+    }
+    return rc;
+}
+
+static bool replace_squeeze_unsqueeze(const std::shared_ptr<Node>& node)
+{
+    auto shape_ps = node->get_output_partial_shape(0);
+    if (shape_ps.rank().get_length() == 0)
+    {
+        return false;
+    }
+    size_t num_unknown = 0;
```

Comment:
No need to introduce the variable  `num_unknown`. Just do the following:
`if (count_unknown_dims(shape_ps) > 1)
    return false;
`